### ISSUE TITLE
Folded heading text not showing in latest vim 8.2 version

### ISSUE THREAD

**Issue by mikeboiko** [masukomi/vim-markdown-folding#42]
I created a simple markdown file to show the differences between vim versions.
`test.md`:
```
# Section 1

a

# Section 2

b
```

Old vim - version
```
:version
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Jun 21 2020 16:11:23)
Included patches: 1-814
Compiled by Arch Linux
```

Old vim - folded buffer shows heading text properly
```
  1 #     Section 1                               [3 lines]---------------------------------------------------------
5   #     Section 2                               [2 lines]---------------------------------------------------------
```

New vim - version
```
:version
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Aug 24 2020 19:37:58)
Included patches: 1-1522
Compiled by Arch Linux
```

New vim - folded buffer doesn't show heading text
```
  1 +--  4 lines folded --------------------------------------------------------------------------------------------
5   +--  3 lines folded --------------------------------------------------------------------------------------------
```

Hopefully this is an easy fix. Thanks!

**Comment by arcdale** [comment]
I have the same problem, since updating to MacVim 8.2

8.2 (2019 Dec 12, compiled Aug 19 2020 06:53:53)
macOS version
Included patches: 1-1456

**Comment by arcdale** [comment]
In 8.2 there seem to be some changes that affect folding.vim

Although I am a developer, I am **not** a vim developer, but this is what I could figure out to get a workaround that seems to  function in 8.2.

The following line no longer sets foldtext

`	let &l:foldtext = s:SID() . 'FoldText()'`

I think this is why we don't see the headings for the folds.

As a workaround, to set foldtext  I first cloned FoldText() to the following since I need access to it at a wider scope than the original definition:-

```
" Workaround for vim 8.2
function FoldTextWorkaround()
    let level = HeadingDepth(v:foldstart)
    let indent = repeat('#', level)
    let title = substitute(getline(v:foldstart), '^#\+\s*', '', '')
    let foldsize = (v:foldend - v:foldstart)
    let linecount = '['.foldsize.' line'.(foldsize>1?'s':'').']'
    return indent.' '.title.' '.linecount
endfunction
```

And then changed 

```
if g:markdown_fold_override_foldtext
  let &l:foldtext = s:SID() . 'FoldText()'
endif
```

to

```
if g:markdown_fold_override_foldtext
  "let &l:foldtext = s:SID() . 'FoldText()'
  setlocal foldtext=FoldTextWorkaround() 
endif
```

With this workaround fold headings in 8.2 look very similar again to how they look in 8.0 and earlier. I'm prepared to use it for now,  but the issue does need looking  at by someone with vim experience.

**Comment by mikeboiko** [comment]
Thanks for the snippet @arcdale!
Since @masukomi doesn't seem to be maintaining this plugin anymore, I made a fork and implemented your bug fix.
For other users experiencing this issue, you can clone my fork:
https://github.com/mikeboiko/vim-markdown-folding

**Comment by rafkaplon** [comment]
This can also be "fixed" by setting `let g:markdown_fold_override_foldtext = 0`. Maybe that was the original intent of that configuration option and the behaviour we're seeing in Vim 8.2 is correct.

**Comment by rafkaplon** [comment]
I believe this is the proper fix:

**Comment by foalford** [comment]
Following the discovery of @arcdale, I did some research and find that [this fix](https://github.com/foalford/vim-markdown-folding/commit/15345cfb919200a74428db0531691fab1147d6a0) will do the job. `:h sid` says it. 

I forked the repo. However writing up a backward compatible fix with proper test is too much for a first time vim script writer. I put it out here for review. And probably come back to finish the rest (comparability and test)

**Comment by jm3** [comment]
@mikeboiko THANK YOU.

**Comment by masukomi** [comment]
as a note: still looking for a new maintainer for this project. If any of you lovely people are interested in taking it over, please let me know.  ;)

**Comment by port19x** [comment]
> Following the discovery of @arcdale, I did some research and find that [this fix](https://github.com/foalford/vim-markdown-folding/commit/15345cfb919200a74428db0531691fab1147d6a0) will do the job. `:h sid` says it.
> 
> I forked the repo. However writing up a backward compatible fix with proper test is too much for a first time vim script writer. I put it out here for review. And probably come back to finish the rest (comparability and test)

Can confirm

**Comment by port19x** [comment]
> as a note: still looking for a new maintainer for this project. If any of you lovely people are interested in taking it over, please let me know. ;)

Since I'm doing the opposite migration to you (from doom emacs back to nvim) I might be able to maintain the plugin at least at a superficial level (reviewing PRs, addressing issues, polishing docs). It's unlikely that I'll ever go deep into vimscript, but if I'm not back on emacs in a week or two I'll shoot you an email.
Using this plugin to emulate org-modes outlining behaviour while writing a paper with markdown + pandoc

**Comment by masukomi** [comment]
fixed with PR #43  (Fixed issue with s:SID in Vim 8.2 )
